### DISCUSSION TITLE
I met some problem : when I use request.send_push_promise, it raises a RuntimeError: Send channel has not been made available

### DISCUSSION THREAD

**Discussion by hotlizimu** [fastapi/fastapi#12107] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request
from fastapi.responses import HTMLResponse
from fastapi.staticfiles import StaticFiles
from fastapi.templating import Jinja2Templates
from fastapi_msgspec.responses import MsgSpecJSONResponse
from fastapi_msgspec.routing import MsgSpecRoute

app = FastAPI(default_response_class=MsgSpecJSONResponse)

app.router.route_class = MsgSpecRoute

app.mount("/static", StaticFiles(directory="static"), name="static")

templates = Jinja2Templates(directory="templates")

async def push_files(request: Request):
    static_files_path = ["/static/images/top.webp", "/static/images/bottom.webp", "/static/images/favicon.ico",
                         "/static/css/base_style.css", "/static/css/iziToast.min.css",
                         "/static/js/base_js.js", "/static/js/iziToast.min.js"]
    [await request.send_push_promise(path) for path in static_files_path]

@app.get("/", response_class=HTMLResponse)
async def get_index(request: Request):
    await push_files(request)
    return templates.TemplateResponse(request=request, name="index.html")
```

### Description

When I run it with hypercorn and visit it on 127.0.0.1:8000, I received a RuntimeError: Send channel has not been made available
![屏幕截图 2024-09-01 204807](https://github.com/user-attachments/assets/56f76e57-c47e-4a2f-af24-3cc2a8af3ca7)
This is my config.toml file:
accesslog = "-"
errorlog = "-"
alpn_protocols = ["h3", "h2", "http/1.1"]
alt_svc_headers = ['h3=":8000"; ma=3600']
quic_bind = "127.0.0.1:8000"
bind = "127.0.0.1:8000"
certfile = "cert.pem"
keyfile = "key.pem"
workers = 1
websocket_ping_interval = 60
I use hypercorn --config config.toml main:app to run it

### Operating System

Windows

### FastAPI Version

0.112.0

### Pydantic Version

2.8.2

### Python Version

3.9

**Comment by sinisaos** [comment]
@hotlizimu I tried your example and found that `fastapi_msgspec` route class make problem. If I commented out (not using the `fastapi_msgspec` route class) everything works fine in my case. Can you try that?
